### PR TITLE
custom db column schema

### PR DESCRIPTION
**Pull request by urusaich**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | no |
| Tests pass? | yes |

What about CustomColumnSchema? Primary for db typecasting.

Allows to do something like this:

Record.php

``` php
public function behaviors()
{
    return [
        'class' => \yii\behaviors\AttributeTypecastBehavior::class,
        'attributeTypes' => [
            'datetime' => function($value){
                if(!$value instanceof \DateTime){
                    $value = \DateTime::createFromFormat(?, $value);
                }

                return $value;
            }
        ],
        'typecastAfterFind' => true,
        'typecastAfterValidate' => false
    ];
}
```

CustomColumnSchema.php

``` php
class ColumnSchema extends \yii\db\ColumnSchema
{
    public function dbTypecast($value)
    {
        if($value instanceof \DateTime){
            return $value->format(?);
        }

        return parent::dbTypecast($value);
    }
}
```

config-db.php

``` php
'db' => [
    'class' => 'yii\db\Connection',
    'schemaMap' => [
        'pgsql' => [
            'class' => \yii\db\pgsql\Schema::class,
            'columnSchemaClass' => \common\CustomColumnSchema::class
        ]
    ]
 ]
```

### CODE REVIEW

File: framework/db/Schema.php [hunk 1]
```diff
@@ -76,6 +76,11 @@
     ];
 
     /**
+     * @var string
```
Comment (samdark) [inline framework/db/Schema.php]: Need a phpdoc here w/ `@since 2.0.11`.

**Review by samdark — CHANGES_REQUESTED** [review]
Overall looks good to me. Need a changelog.

**Comment by nanodesu88** [comment]
updated

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Perfect. If you'll merge your branch with current master I'll merge pull request right away.

**Comment by nanodesu88** [comment]
Solved?

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Yes. Thank you!

**Pull request merged**